### PR TITLE
Integrate scala examples with the integration tests

### PR DESCRIPTION
**Pull request by tdas**
Update `run-integration-tests.py` to run the scala tests
- Update scala example build file to take the version from the env variable DELTA_VERSION
- Update run-integration-tests.py to call all scala examples with the given version injected via env variable

### CODE REVIEW

File: run-integration-tests.py [hunk 1]
```diff
@@ -26,35 +26,56 @@
...
+    print("##### Running Python tests on version %s #####" % str(version))
+    clear_artifact_cache()
     test_dir = path.join(root_dir, path.join("examples", "python"))
     test_files = [path.join(test_dir, f) for f in os.listdir(test_dir)
                   if path.isfile(path.join(test_dir, f)) and
                   f.endswith(".py") and not f.startswith("_")]
     python_root_dir = path.join(root_dir, "python")
     extra_class_path = path.join(python_root_dir, path.join("delta", "testing"))
-    # bintray repo url
+    package = "io.delta:delta-core_2.11:" + version
```
Comment (rahulsmahadev) [inline run-integration-tests.py:57]: should we have a separate `get_scala_version` method to when we eventually move to 2.12

**Comment by rahulsmahadev** [comment]
One comment but LGTM!

File: run-integration-tests.py [hunk 2]
```diff
@@ -26,35 +26,56 @@
...
+                  if f.endswith(".scala") and not f.startswith("_")]
+    with WorkingDirectory(test_dir):
+        for test_class in test_classes:
+            try:
+                cmd = ["build/sbt", "runMain example.%s" % test_class]
+                print("Running Scala tests in %s\n=====================" % test_class)
+                print("Command: %s" % str(cmd))
+                run_cmd(cmd, stream_output=True, env={"DELTA_VERSION": str(version)})
+            except:
+                print("Failed Scala tests in %s" % (test_class))
```
Comment (pranavanand) [inline run-integration-tests.py:44]: Will this print the original error as well?

Comment (tdas) [reply run-integration-tests.py:44]: yes. raise below throws the original exception.

**Pull request closed without merging**